engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -1068,12 +1071,48 @@ private void finishApplication(FinalApplicationStatus finalStatus) throws YarnEx
    * Check for containers that were allocated in a previous attempt.
    * If the containers are still alive, wait for them to check in via heartbeat.
    */
-  private void previouslyAllocatedContainers(List<Container> containers)
+  private void previouslyAllocatedContainers(List<Container> containersListByYarn, Collection<ContainerId> releasedContainers)
   {
-    for (Container container : containers) {
-      this.allocatedContainers.put(container.getId().toString(), new AllocatedContainer(container));
-      //check the status
-      nmClient.getContainerStatusAsync(container.getId(), container.getNodeId());
+    LOG.info("Finding information about the previously allocated containers.");
+
+    if (containersListByYarn.size() != 0) {
+
+      LOG.info("YARN has the list of {} running containers.", containersListByYarn.size());
+      LOG.info("Number of containers from StreamingContainerManager is {}", dnmgr.getPhysicalPlan().getContainers().size());
+
+      Map<String, Container> fromYarn = new HashMap<>();
+      Set<String> fromStreamingContainerManager = new HashSet<>();
+
+      for (Container container : containersListByYarn) {
+        fromYarn.put(container.getId().toString(), container);
+      }
+
+      for (PTContainer ptContainer : dnmgr.getPhysicalPlan().getContainers()) {
+        fromStreamingContainerManager.add(ptContainer.getExternalId());
+      }
+
+      for (PTContainer ptContainer : dnmgr.getPhysicalPlan().getContainers()) {
+
+        String containerId = ptContainer.getExternalId();
+        Container container = fromYarn.get(containerId);
+
+        if (container != null) {
+          allocatedContainers.put(containerId, new AllocatedContainer(container));
+          fromYarn.remove(containerId);
+          fromStreamingContainerManager.remove(containerId);
+        }
+      }
+
+      for (Container container : fromYarn.values()) {
+        releasedContainers.add(container.getId());
```

Comment:
IMO, this is what needs to be logged at INFO level. There is Yarn container allocated that does not have a corresponding container in the physical plan.